classes/deploy_target.py

```diff
@@ -0,0 +1,123 @@
+from abc import ABC, abstractmethod
+import re
+
+from classes.controller_methods import ControllerMethods
+
+class DeployTarget(ControllerMethods, ABC):
+ 
+    def __init__(self):
+        self.controllerIps = []
+
+   
+    def parse_nile(self):
+        """ Parses a Nile intent from text and return dictionary with intent operation targets """
+        from_pattern = re.compile(r".*from (endpoint|service)(\(\'.*?\'\)).*")
+        to_pattern = re.compile(r".*to (endpoint|service)(\(\'.*?\'\)).*")
+        target_pattern = re.compile(r".*for ((endpoint|service|group|traffic)(\(\'.*?\'\))(, (endpoint|service|group|traffic)(\(\'.*?\'\)))*).*")
+        set_unset_pattern = re.compile(r".*(set|unset) ((quota|bandwidth)(\(\'.*?\'\))(, (quota|bandwidth)(\(\'.*?\'\)))*).*")
+        allow_block_pattern = re.compile(r".*(allow|block) ((traffic|service|protocol)(\(\'.*?\'\))(, (traffic|service|protocol)(\(\'.*?\'\)))*).*")
+        add_remove_pattern = re.compile(r".*(add|remove) ((middlebox)(\(\'.*?\'\))(, (middlebox)(\(\'.*?\'\)))*).*")
+        start_pattern = re.compile(r".*start (hour|datetime|timestamp)(\(\'.*?\'\)).*")
+        end_pattern = re.compile(r".*end (hour|datetime|timestamp)(\(\'.*?\'\)).*")
+
+        op_targets = {
+            'operations': [],
+            'targets': []
+        }
+
+        result = from_pattern.search(self)
```

Comment:
In general, avoid passing the entire self object as a param. 
There are cases where this is necessary, but they are very rare. This is not one of them.



```suggestion
        result = from_pattern.search(intent)
```

The same applies to the other cases below.